singer_sdk/streams/core.py

```diff
@@ -1166,7 +1175,7 @@ def _sync_records(  # noqa: C901
                         )
                         raise
 
-                    if selected:
+                    if not self.__abstract__ and selected:
```

Comment:
```suggestion
                    if not selected:
```

Reply:
Personally, I would remove the redundant assignment

```py
        selected = self.selected
```

above for clarity. Then this becomes

 ```suggestion
                    if not self.selected:
```

as below.